compiler+runtime/src/cpp/jank/runtime/obj/lazy_sequence.cpp

```diff
@@ -62,18 +70,16 @@ namespace jank::runtime::obj
 
   lazy_sequence_ptr lazy_sequence::next_in_place()
   {
-    resolve_seq();
-    if(sequence)
+    // sequence is non-nullptr
+    auto const n(runtime::next_in_place(sequence));
+    if(n == nil::nil_const())
     {
-      /* We don't know we own this sequence, so we can't use next_in_place. */
-      sequence = runtime::next(sequence);
-      if(sequence == nil::nil_const())
-      {
-        sequence = nullptr;
-      }
-      return sequence ? this : nullptr;
+      // seq is consumed, no need to update
+      //sequence = nullptr;
```

Comment:
Let's not leave commented code in to explain why we don't need it. The comment alone will suffice, if we make it clear enough. This comment is nowhere near clear enough, since the removal of this one line has huge implications.

It's safe to not set `sequence` to null, we're saying, because we know that we're in a fresh sequence. Nobody else has a pointer to this sequence. Thus, if we fully consume `sequence` and we're done with our current lazy seq, we return null and expect that this sequence will never be used again.